data-prepper-plugins/geoip-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/geoip/provider/MaxMindGeoIpProvider.java

```diff
@@ -0,0 +1,62 @@
+package com.amazon.dataprepper.plugins.prepper.geoip.provider;
+
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.plugins.prepper.geoip.GeoIPPrepperConfig;
+import com.maxmind.db.CHMCache;
+import com.maxmind.geoip2.DatabaseReader;
+import com.maxmind.db.InvalidDatabaseException;
+import com.maxmind.geoip2.exception.AddressNotFoundException;
+import com.maxmind.geoip2.exception.GeoIp2Exception;
+import com.maxmind.geoip2.model.CityResponse;
+import com.maxmind.geoip2.record.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Objects;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class MaxMindGeoIpProvider implements GeoIpProvider {
+    private final String databasePath;
+    private final DatabaseReader databaseReader;
+    private final File database;
+    private static final Logger LOG = LoggerFactory.getLogger(MaxMindGeoIpProvider.class);
+
+    public MaxMindGeoIpProvider(PluginSetting pluginSetting) throws IllegalArgumentException{
```

Comment:
Also it might be better to have the constructor take in explicit values instead of just a PluginSetting object. Let the factory be responsible for pulling out the needed fields (like DATABASE_PATH).

It'll make your unit tests easier because you won't have to create a PluginSetting each time.